### PR TITLE
ensure driver always gets info about testrun (>=FF58)

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: src/browser/connection/index.js [hunk 1]
```diff
@@ -115,6 +116,18 @@ export default class BrowserConnection extends EventEmitter {
         }, this.HEARTBEAT_TIMEOUT);
     }
 
+    async _getTestRunUrl (isTestDone) {
```
Comment (AndreyBelym) [inline src/browser/connection/index.js]: I fell we can simplify logic here and in `getStatus`.

Let's make `BrowserConnection` constructor this way:
```
class BrowserConnection {
    constructor () {
        ...
        this.pendingTestRunUrl = null;
```
Then let's make `_getTestRunUrl` this way:
```
     async  _popNextTestRunUrl () {
        while (this.hasQueuedJobs && !this.currentJob.hasQueuedTestRuns)
            this.jobQueue.shift();

        this.pendingTestRunUrl = this.hasQueuedJobs ? await this.currentJob.popNextTestRunUrl(this) : null;  
    }
    
    async _getTestRunUrl (isRunningBrowser) {
        if (isRunningBrowser || !this.pendingTestRunUrl)
            this._popNextTestRunUrl();

        return this.pendingTestRunUrl;
    }
```
Then we can write `getStatus` in this way:
```
async getStatus (isRunningBrowser) {
    if (!this.idle && !isRunningBrowser) {
        this.idle = true;
        this.emit('idle');    
    }

    if (this.opened) {
        var testRunUrl = await this._getTestRunUrl(isRunningBrowser);

        if (testRunUrl) {
            this.idle = false;
            return { cmd: COMMAND.run, url: testRunUrl };
        }
    }

    return { cmd: COMMAND.idle, url: this.idleUrl };
}
```
\cc @AlexanderMoskovkin

Comment (AlexKamaev) [reply src/browser/connection/index.js]: Didn't want to cache any data in _popNextTestRunUrl. This function should only get testRun without any other actions. I tried to move all logic to one function _popNextTestRunUrl without any checks on other levels

Comment (AndreyBelym) [reply src/browser/connection/index.js]: That's right, I've updated my code.

Comment (AlexKamaev) [reply src/browser/connection/index.js]: I agree, It looks better, but I would prefer not to rename param to isRunningBrowser

**Pull request merged**